src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigToJobMapper.java

```diff
@@ -106,6 +98,31 @@ public static FlowDefinition mapBuildConfigToFlow(BuildConfig bc) throws IOExcep
 		}
 	}
 
+	public static GitSCM getGitSCM(BuildSource source, BuildConfig bc){
+    GitBuildSource gitSource = source.getGit();
+    String branchRef = gitSource.getRef();
+    List<BranchSpec> branchSpecs = Collections.emptyList();
+    GitSCM gitSCM = null;
+    String credentialsId = null;
+    LocalObjectReference sourceSecret = source.getSourceSecret();
+    if (sourceSecret != null){
+      try{
+        credentialsId = updateSourceCredentials(bc);
+      } catch (Exception e){
+        LOGGER.warning("Encountered "+e.toString()+" during updateSourceCredentials()");
+      }
+    }
+    if (isNotBlank(branchRef)) {
+      branchSpecs = Collections.singletonList(new BranchSpec(branchRef));
+    }
+    // if credentialsID is null, go with an SCM where anonymous has to be sufficient
+    gitSCM = new GitSCM(
+        Collections.singletonList(new UserRemoteConfig(gitSource.getUri(), null, gitSource.getRef(), credentialsId)),
```

Comment:
gitSource.getRef() has already been set in a variable previously. Better to you the declared variable.